server/impl/meetup.go

```diff
@@ -1,15 +1,762 @@
+		)
+	SELECT * FROM all_tags
+	`, strings.Join(placeholders, ", ")), variables...).Scan(&dbMeetup.Tags).Error
+	} else {
+		err = tx.Model(dbMeetup).Association("Tags").Clear()
+		dbMeetup.Tags = nil
+	}
+	return err
+}
+
+func (i *Implementation) modelMeetupToDBMeetup(dbMeetup *db.Meetup, modelMeetup *models.Meetup) error {
+	dbMeetup.Title = modelMeetup.Title
+	dbMeetup.Description = modelMeetup.Description
+	dbMeetup.MaxCapacity = swag.Int64Value(modelMeetup.MaxCapacity)
+	dbMeetup.MinCapacity = swag.Int64Value(modelMeetup.MinCapacity)
+	dbMeetup.Description = modelMeetup.Description
+	dbMeetup.Owner, _ = db.UserIDFromString(string(modelMeetup.Owner))
+	dbMeetup.Time = time.Time(modelMeetup.Time)
+
+	if modelMeetup.Location != nil {
+		var coordinates db.Coordinates
+		if modelMeetup.Location.Coordinates != nil {
+			coordinates = db.Coordinates{
+				Lat: modelMeetup.Location.Coordinates.Lat,
+				Lon: modelMeetup.Location.Coordinates.Lon,
+			}
+		}
+		dbMeetup.Location = db.MeetupLocation{
+			Coordinates: coordinates,
+			Name:        modelMeetup.Location.Name,
+			URL:         modelMeetup.Location.URL,
+		}
+	}
+	return nil
+}
+
+func modelMeetupRequestBodyToModelMeetup(modelMeetupRequestBody *models.MeetupRequestBody, id string) models.Meetup {
+	var modelMeetup models.Meetup
+	modelMeetup.Tags = modelMeetupRequestBody.Tags
+	modelMeetup.Owner = models.UserID(id)
+	modelMeetup.MinCapacity = modelMeetupRequestBody.MinCapacity
+	modelMeetup.MaxCapacity = modelMeetupRequestBody.MaxCapacity
+	modelMeetup.Time = modelMeetupRequestBody.Time
+	modelMeetup.Title = modelMeetupRequestBody.Title
+	modelMeetup.Description = modelMeetupRequestBody.Description
+	if modelMeetupRequestBody.Location != nil && modelMeetupRequestBody.Location.Coordinates != nil {
+		coordinates := &models.Coordinates{
+			Lat: modelMeetupRequestBody.Location.Coordinates.Lat,
+			Lon: modelMeetupRequestBody.Location.Coordinates.Lon,
+		}
+		modelMeetup.Location = &models.Location{
+			Coordinates: coordinates,
+			Name:        modelMeetupRequestBody.Location.Name,
+			URL:         modelMeetupRequestBody.Location.URL,
+		}
+	}
+	return modelMeetup
+}
+
+func dbMeetupToModelMeetup(dbMeetup *db.Meetup, userID string) *models.Meetup {
+	location := &models.Location{}
+	if dbMeetup.Location.Coordinates.Lat != nil && dbMeetup.Location.Coordinates.Lon != nil {
+		coordinates := &models.Coordinates{
+			Lat: dbMeetup.Location.Coordinates.Lat,
+			Lon: dbMeetup.Location.Coordinates.Lon,
+		}
+		location.Coordinates = coordinates
+	}
+	if dbMeetup.Location.URL != "" {
+		location.URL = dbMeetup.Location.URL
+	}
+	if dbMeetup.Location.Name != "" {
+		location.Name = dbMeetup.Location.Name
+	}
+
+	// Determine if user was rejected or not
+	var rejected bool
+	if dbMeetup.Attendees != nil && userID != "" {
+		for _, attendee := range dbMeetup.Attendees {
+			if attendee.IDString() == userID {
+				rejected = false
+				break
+			}
+		}
+		rejected = true
```

Comment:
Just because the user is not in the list of attendees doesn't mean that they were rejected. I believe you should check `dbMeetup.RejectedAttendees` instead.